contents/docs/plugins/build/overview.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Overview
+sidebar: Docs
+showTitle: true
+---
+
+A great way to understand what you can do with plugins is to understand how data flows throw plugins.
+
+There's 2 big concepts to remember:
+
+1. Every plugin acts on a single event coming in.
+
+2. Plugins act on events before they are stored.
+
+Each plugin can add more information to the event, or even modify existing properties. For example, the [GeoIP plugin](https://posthog.com/plugins/geoip) is a plugin that adds GeoIP information to the event.
+
+Note that the output of one plugin goes into the next plugin. Here's how this looks:
+
+![GeoIP Plugin Example](../../../images/plugins/geoip-plugin-example.png)
+
+However, this isn't all. Plugins _don't_ have to modify events at all: they can do other things when an event comes in.
+
+For example, you can send an event to AWS S3 whenever you see it in PostHog. Indeed, the [S3 plugin](https://posthog.com/plugins/s3-export) does exactly that. In this case, it doesn't matter if the S3 export succeeds or not, the event will always be stored.
+
+![S3 Plugin Example](../../../images/plugins/s3-plugin-example.png)
+
+There's also some plugins that enqueue jobs to run in the future. More information about jobs in [the developer reference](./reference#jobs-1).
+
+Now, how do you make all of this happen? Every plugin is two files: `index.js` and `plugin.json`. The index file has code for the entire plugin, and the json file has configuration for user inputs. This config is what you see in the PostHog UI:
+
+![Plugin Configuration Example](../../../images/plugins/plugin-configuration.png)
+
+We have some special function names that allow you to process an event, like in the GeoIP plugin, or to do something else entirely, like in the S3 export plugin. We expect `index.js` to export these special functions.
+
+Two notable ones are `processEvent` and `onEvent`. Both of them take in a single event and the meta object. More details on the meta object [in the developer reference](./reference#pluginmeta), but one key property is `meta.config`, which allows your code to read the configuration values set by users. Yes, the same configuration you set via `plugin.json`.
+
+If you want to add new properties to your event, like in the GeoIP Plugin, you'd use the `processEvent` function. Here's a sample plugin that adds the `hello` property to the event.
+
+```js
+/* Runs on every event */
+export function processEvent(event, meta) {
+    // Some events (like $identify) don't have properties
+    if (event.properties) {
+        event.properties['hello'] = `Hello ${meta.config.name || 'world'}`
+    }
+    // Return the event to ingest, return nothing to discard  
+    return event
+}
+```
+
+Note how you need to return the event to ensure the chain continues. If you return `null` or `undefined`, you're telling us to discard this event.
```

Comment:
Could add an example to the schema enforcer here. 

"This could be a deliberate decision, for example to prevent the ingestion of [events that don't fit a certain schema](/plugins/schema-enforcer)"